unmarshaler.go

```diff
@@ -252,6 +285,7 @@ func (d *decoder) scopeWithArrayTable(x target, key ast.Iterator) (target, bool,
 		x, err = scopeSlice(true, x)
 	case reflect.Array:
 		x, err = d.scopeArray(true, x)
+	default:
```

Comment:
That doesn't seem useful 😄

Reply:
indeed... just to satisfy the linter. I hesitated between nolint and this, but i guess it's less bad this way :)